### PR TITLE
[bazel/Mask ROM] Add a regression tests for CI

### PR DESCRIPTION
**Pull request by drewmacrae**
Invoke regression tests with bazel to protect artifacts in the Mask ROM
Only fail if targets begin to fail

Signed-off-by: Drew Macrae <drewmacrae@google.com>

### CODE REVIEW

File: ci/scripts/mask-rom-tests.sh [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+set -e -x
+
+# This doesn't install dependencies, but should otherwise behave the
+# same as what CI would do on a pull request
+DEFAULT="master"
+
+# use tests from a condonedlist
```
Comment (mcy) [inline ci/scripts/mask-rom-tests.sh:12]: What's the criteria for a test being on this list? Can we file an issue to get rid of the list altogether once we delete Meson?

**Comment by drewmacrae** [comment]
I still need to check out master to discover the merge base to know what was previously broken.

**Comment by drewmacrae** [comment]
[The logs](https://dev.azure.com/lowrisc/4c50229e-f6cb-4f39-8522-585bb3841438/_apis/build/builds/72642/logs/138) show 5 passing tests that also pass on the merge base so this PR passes. It reruns at the merge base because Bazel ran into trouble when building a number of tests.

It's configured to pass when it's merged in regardless of what happens with merge skew etc. (there's no merge base to compare against and I can't think of a good way to discover which commits are in the PR when it's merged with a linear history) It will print issues to the log if it has any.

**Comment by drewmacrae** [comment]
Chris has a framework to turn up more of the tests by fixing them here: https://github.com/lowRISC/opentitan/pull/11416
Meson makes it hard to submit these for now, but this CI should run the newly enabled tests once this is submitted

**Review by mcy — APPROVED** [review]

**Review by milesdai — APPROVED** [review]

**Review by arunthomas — APPROVED** [review]

**Comment by drewmacrae** [comment]
I'm discussing an issue with this commit in https://opentitan.slack.com/archives/C0104UQPJG1/p1647912170822099 the slack #general channel. I have a fix here https://github.com/lowRISC/opentitan/pull/11588 and incoming/pending PRs should not be affected, but postmerge tests are broken.

**Pull request merged**